core_backend/app/contents/routers.py

```diff
@@ -158,6 +165,209 @@ async def retrieve_content_by_id(
     return _convert_record_to_schema(record)
 
 
+@router.post("/csv-upload", response_model=List[ContentRetrieve])
+async def bulk_upload_contents(
+    file: UploadFile,
+    user_db: Annotated[UserDB, Depends(get_current_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> List[ContentRetrieve] | None:
+    """
+    Upload, check, and ingest contents in bulk from a CSV file.
+
+    Note: If there are any issues with the CSV, the endpoint will return a 400 error
+    with the list of issues under detail in the response body.
+    """
+
+    # TODO: deal with tags!
+
+    # Ensure the file is a CSV
+    if file.filename is None or not file.filename.endswith(".csv"):
+        error_list_model = CustomErrorList(
+            errors=[
+                CustomError(
+                    type="invalid_format",
```

Comment:
Since we are only dealing with contents, I think we should use `title` instead of `content_title`. It would make it more user friendly and would lead to less errors (For example `content-title` instead of `content_title`). Same for content_text

Reply:
I chose this as per Tanmay's design and also how our export data looks like, but I do agree. I think in a later PR we should rename our endpoint response, export function, and this import function.